src/file_utils.rs

```diff
@@ -104,3 +104,59 @@ pub(crate) fn decode_resource_data_entry(yaml_location: &YamlLocation, value_at_
     }
     .clone())
 }
+
+pub(crate) fn add_annotation(resource: &mut Value, yaml_location: &YamlLocation) -> Result<()> {
+    if resource.pointer_mut("/metadata/annotations").is_none() {
+        resource
+            .pointer_mut("/metadata")
+            .context("metadata must exist")?
+            .as_object_mut()
+            .context("metadata must be an object")?
+            .insert(
+                String::from("annotations"),
+                Value::Object(serde_json::Map::new()),
+            );
+    }
+
+    let current_value_string = match resource.pointer_mut("/metadata/annotations/recert-edited") {
+        Some(annotation_data) => {
+            annotation_data.as_str().context("recert annotation data must be a string")?
+        },
+        None => "{}"
+    };
+
+    let mut annotation_value: Value = serde_json::from_str(current_value_string).context("recert annotation data must be valid json")?;
+
+    let edited_index = match &yaml_location.value {
+        LocationValueType::Pem(pem_info) => {
+            Value::String(pem_info.pem_bundle_index.to_string())
+        },
+        _ => {
+            Value::String(String::from("N/A"))
+        },
+    };
+
+    match annotation_value.get_mut(&yaml_location.json_pointer) {
+        Some(locations) => {
+            locations
+                .as_array_mut()
+                .context("locations must be an array")?
+                .push(edited_index);
```

Comment:
Can we sort the indices for a bit of extra determinism?

Reply:
Sure, but given that I went with strings (to accommodate non-PEM cases) it might sort strangely with indexes over 9.

Reply:
So I looked into this and because the elements at this point are `Values` (which don't have `Ord`) it would be a bit of a pain to sort for it to only still come out kind of wrong.

I can still obviously do it (and probably handle the numeric/"N/A" cases too since I'd have to implement the comparison myself), just questioning if it's worth the effort.

Reply:
For now let's leave it as is, maybe we'll improve it later. I think maybe even compact "0","1","2","3" into "0-3" since the most common case is if one cert was modified, it's likely all other certs in the bundle have also been modified. I don't want our annotation to look too ugly and take up screen space (like managed-fields usually does)